File: couch/csharp/create/couch_create.cs
// ------------------------------------------------------------------
//	couch_create.cs
//
//					Jul/24/2014
//
// ------------------------------------------------------------------
using System;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

using Newtonsoft.Json;
// ------------------------------------------------------------------
public class couch_create
{
// ------------------------------------------------------------------
static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string url_collection  = "http://localhost:5984/nagano";

	string user = "";
	string password = "";

	Console.WriteLine (url_collection);

	try
		{
		get_uri.rest_delete_proc (url_collection,user,password);
		}
	catch (System.Net.WebException ee)
		{
		Console.WriteLine ("*** error *** rest_delete_proc ***");
		Console.WriteLine (ee);
		}

	try
		{
		get_uri.put_uri_string_proc (url_collection,"",user,password);
		}
	catch (System.Net.WebException ee)
		{
		Console.WriteLine ("*** error *** put_uri_string_proc ***");
		Console.WriteLine (ee);
		}

	Dictionary <string,Object> dict_aa = data_prepare_proc ();

	foreach (KeyValuePair<string, Object> kv in dict_aa)
		{
		Dictionary <string,string> unit_aa
				= (Dictionary <string,string>)kv.Value;

		string str_json = JsonConvert.SerializeObject (unit_aa);

		string url_target = url_collection + "/" + kv.Key;
		get_uri.put_uri_string_proc (url_target,str_json,user,password);
		}
	
	Console.WriteLine ("*** 終了 ***");
}

// ------------------------------------------------------------------
static Dictionary <string,Object> data_prepare_proc ()
{
	Dictionary <string,Object> dict_aa
			= new Dictionary <string,Object> ();

	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2021","長野",34512,"2009-6-7");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2022","松本",17954,"2009-5-19");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2023","上田",76928,"2009-3-28");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2024","小諸",25197,"2009-12-21");

	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2025","岡谷",14976,"2009-11-25");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2026","塩尻",38945,"2009-1-26");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2027","茅野",96184,"2009-1-21");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2028","飯田",51632,"2009-3-7");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2029","中野",64823,"2009-2-12");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2030","諏訪",52715,"2009-1-14");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2031","駒ヶ根",84329,"2009-8-9");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2032","佐久",28618,"2009-12-15");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2033","伊那",87231,"2009-12-21");
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2034","千曲",71489,"2009-2-24");

	return	dict_aa;
}

// ------------------------------------------------------------------
}
// ------------------------------------------------------------------

File: couch/fsharp/create/couch_create.fs
// ------------------------------------------------------------------
//	couch_create.fs
//
//					Mar/03/2016
//
// ------------------------------------------------------------------
open System
open System.Text
open Microsoft.FSharp.Core
open System.IO
open System.Net
open System.Collections
open System.Collections.Generic

open Newtonsoft.Json
// -------------------------------------------------------------------
let data_prepare_proc () =

  let dict_aa = new Dictionary<string,Object>()


  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2021","長野",27613,"1997-5-7")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2022","松本",36179,"1997-2-12")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2023","上田",84326,"1997-3-28")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2024","小諸",79583,"1997-2-21")

  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2025","岡谷",51978,"1997-1-5")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2026","塩尻",35647,"1997-5-22")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2027","茅野",45786,"1997-7-17")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2028","飯田",14965,"1997-3-4")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2029","中野",71324,"1997-10-12")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2030","諏訪",32845,"1997-6-9")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2031","駒ヶ根",81576,"1997-5-7")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2032","佐久",58124,"1997-2-12")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2033","伊那",34897,"1997-8-21")
  let dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2034","千曲",21746,"1997-1-25")

  dict_aa

// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =

  printfn "*** 開始 ***"


  let url_collection = "http://localhost:5984/nagano"

  let user = ""
  let password = ""

  let _ = get_uri.rest_delete_proc (url_collection,user,password)
  let _ = get_uri.put_uri_string_proc (url_collection,"",user,password)

  let dict_aa = data_prepare_proc ()

  printfn "%d" dict_aa.Count

  for unit in dict_aa do
//    Console.WriteLine (unit.Key)
    let str_json_out = JsonConvert.SerializeObject (unit.Value)
//    printfn "%s" str_json_out
    let url = url_collection + "/" + unit.Key
//    printfn "%s" url
    get_uri.put_uri_string_proc (url,str_json_out,user,password)
  done


  printfn "*** 終了 ***"

  0

// ------------------------------------------------------------------
